.github/actions/connect-arc/action.yml

```diff
@@ -0,0 +1,61 @@
+# Copyright (c) Microsoft. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+# This workflow assumes your cluster can be reached by the agent this workflow runs on.
+
+name: 'Connect your cluster to Azure ARC'
+description: |
+  Action to connect your kubernetes cluster to Azure ARC using `az connectedk8s connect`. 
+  Will add the connectedk8s Azure CLI extension if not already installed.
+inputs:
+  cluster-name:
+    required: true
+    description: 'Cluster Name'
+  resource-group:
+    required: true
+    description: 'Resource Group'
+  config:
+    description: Contents of kubeconfig to connect to cluster. If not provided, `~/.kube/config` will be used.
+  custom-locations-oid:
+    description: |
+      Object ID of Custom Locations Application in your tenant. 
+      Find with: `az ad sp show --id bc313c14-388c-4e7d-a58e-70017303ee3b --query id -o tsv`.
+      
+      Please provide this value if your azure login context lacks permission to make this call.
+
+runs:
+  using: 'composite'
+  steps:
+    - name: "Install connectedk8s extension"
+      run: az extension add -n connectedk8s -y
+      shell: bash
+    - name: "Write local kubeconfig"
+      if: inputs.config
+      env:
+        config: |
+          ${{ inputs.config }}
+      run: |
+        CONFIGPATH=$HOME/.kube/config
+        mkdir -p $HOME/.kube
+        echo "${{ env.config }}" > $CONFIGPATH
+        chmod 644 $CONFIGPATH
+        echo "ARC_CLUSTER_CONFIG=$CONFIGPATH" >> $GITHUB_ENV
+      shell: bash
+    - name: "Connect cluster to ARC and enable features"
+      env:
+        cluster-name: "${{ inputs.cluster-name }}"
+        resource-group: "${{ inputs.resource-group }}"
+        custom-locations-oid: ${{ inputs.custom-locations-oid && format('--custom-locations-oid {0}', inputs.custom-locations-oid) || '' }}
+      run: >-
+        az connectedk8s connect
+        -n ${{ env.cluster-name }}
+        -g ${{ env.resource-group }}
+        ${{ env.custom-locations-oid }}
+        ${{ env.ARC_CLUSTER_CONFIG && format('--kube-config {0}', env.ARC_CLUSTER_CONFIG) || '' }}
+        
+        az connectedk8s enable-features --features custom-locations cluster-connect
+        -n ${{ env.cluster-name }}
+        -g ${{ env.resource-group }}
+        ${{ env.custom-locations-oid }}
+        ${{ env.ARC_CLUSTER_CONFIG && format('--kube-config {0}', env.ARC_CLUSTER_CONFIG) || '' }}
```

Comment:
Same as lines 53-54 -> is there a point for these values when they have no argument?

Reply:
Effectively no, but rather than a giant `if-else` string builder, the `>-` yaml block scalar will turn each line that evaluates into an empty string as a single space ` `. This way we can keep the same command string, but if values are not present in input, they will only result in a single extra space between params in the final command:
`az connectedk8s ... -g [myRG]  --kube-config ~/.kube/config`

This was a readability decision, technically we could put all the configs on one line:
`-g ${{ env.resource-group }}${{ env.custom-locations-oid }}${{ env.ARC_CLUSTER_CONFIG && format('--kube-config {0}', env.ARC_CLUSTER_CONFIG) || '' }}` but that's gonna be hellish to maintain

Reply:
jk I am stupid since I see that 54 + 60 have the kube config arg... however the custom locations oid does not...

Reply:
Correct, and that's actually a limitation of github environment variables not being able to reference other environment variables (even if defined in other steps) unless they're defined as outputs.

There is a solution to pipe the `ARC_CLUSTER_CONFIG` var as an output instead of an environment variable, but for simplicity I just expanded it inline instead of adding step outputs / needs / requirements to the other steps.